catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java

```diff
@@ -227,6 +239,24 @@
     public String getId();
 
     /**
+     * Returns the tags associated with this metacard.
+     *
+     * @return Returns the TAGS attribute values if it exists otherwise return null
+     */
+    public default Set<String> getTags() {
+        Attribute attribute = getAttribute(TAGS);
+        if (attribute == null || attribute.getValue() == null) {
+            return null;
+        } else {
+            Set<String> tags = new HashSet<>();
```

Comment:
I missed the type change from Serializable to String.  The loop is fine but you could also do it with a stream.

```        
getAttribute(TAGS).getValues()
                .stream()
                .map(String::valueOf)
                .collect(Collectors.toList());```